src/modules/hip/kernel/lens_correction.hpp

```diff
@@ -0,0 +1,158 @@
+    float radialCoeff[6] = {distortionCoeffs_f8.f1[0], distortionCoeffs_f8.f1[1], distortionCoeffs_f8.f1[4], distortionCoeffs_f8.f1[5], distortionCoeffs_f8.f1[6], distortionCoeffs_f8.f1[7]};
+    float tangentialCoeff[2] = {distortionCoeffs_f8.f1[2], distortionCoeffs_f8.f1[3]};
+
+    uint dstIdx = id_z * remapTableStridesNH.x + id_y * remapTableStridesNH.y + id_x;
+    d_float8 locDst_f8x;
+    locDst_f8x.f4[0] = static_cast<float4>(id_x) + make_float4(0, 1, 2, 3);
+    locDst_f8x.f4[1] = static_cast<float4>(id_x) + make_float4(4, 5, 6, 7);
+
+    float4 one_f4 = static_cast<float4>(1.0f);
+    float4 two_f4 = static_cast<float4>(2.0f);
+    d_float8 z_f8, y_f8, x_f8;
+    camera_coordinates_hip_compute(&z_f8, id_y, &locDst_f8x, &inverseMatrix_f9.f3[2]);          // float zCamera = id_y * inverseMatrix.f1[7] + inverseMatrix.f1[8] + id_x * inverseMatrix.f1[6]
+    camera_coordinates_hip_compute(&y_f8, id_y, &locDst_f8x, &inverseMatrix_f9.f3[1]);          // float yCamera = id_y * inverseMatrix.f1[4] + inverseMatrix.f1[5] + id_x * inverseMatrix.f1[3]
+    camera_coordinates_hip_compute(&x_f8, id_y, &locDst_f8x, &inverseMatrix_f9.f3[0]);          // float xCamera = id_y * inverseMatrix.f1[1] + inverseMatrix.f1[2] + id_x * inverseMatrix.f1[0]
+    rpp_hip_math_divide8_const(&z_f8, &z_f8, one_f4);                                           // float z = 1./zCamera
+    rpp_hip_math_multiply8(&y_f8, &z_f8, &y_f8);                                                // float y = yCamera * z;
+    rpp_hip_math_multiply8(&x_f8, &z_f8, &x_f8);                                                // float x = xCamera * z;
+
+    d_float8 ySquare_f8, xSquare_f8;
+    rpp_hip_math_multiply8(&y_f8, &y_f8, &ySquare_f8);                                          // float ySquare = x * x
+    rpp_hip_math_multiply8(&x_f8, &x_f8, &xSquare_f8);                                          // float xSquare = x * x
+
+    d_float8 r2_f8, kr_f8, kr1_f8, kr2_f8;
+    rpp_hip_math_add8(&xSquare_f8, &ySquare_f8, &r2_f8);                                        // float r2 = xSquare + ySquare
+
+    d_float8 r2Cube_f8, r2Square_f8;
+    rpp_hip_math_multiply8(&r2_f8, &r2_f8, &r2Square_f8);                                       // float r2Square = r2 * r2;
+    rpp_hip_math_multiply8(&r2Square_f8, &r2_f8, &r2Cube_f8);                                   // float r2Cube = r2Square * r2;
+
+    // float kr = (1 + (radialCoeff[2] * r2Cube) + (radialCoeff[1] * r2Square) + (radialCoeff[0]) * r2)) / (1 + (radialCoeff[5] * r2Cube) + (radialCoeff[4] * r2Square) + (radialCoeff[3]) *r2))
+    kr1_f8.f4[0] = (one_f4 + static_cast<float4>(radialCoeff[2]) * r2Cube_f8.f4[0] + static_cast<float4>(radialCoeff[1]) *  r2Square_f8.f4[0] + static_cast<float4>(radialCoeff[0]) *  r2_f8.f4[0]);
+    kr1_f8.f4[1] = (one_f4 + static_cast<float4>(radialCoeff[2]) * r2Cube_f8.f4[1] + static_cast<float4>(radialCoeff[1]) *  r2Square_f8.f4[1] + static_cast<float4>(radialCoeff[0]) *  r2_f8.f4[1]);
```

Comment:
To avoid this many static_cast between L90-L94, can't we assign a single d_float24 radialCoeff_f24 on L61 as 6 float4s?
Something like this will work?
```
d_float24 radialCoeff_f24 = {static_cast<float4>(distortionCoeffs_f8.f1[0]),
                                              static_cast<float4>(distortionCoeffs_f8.f1[1]),
                                              static_cast<float4>(distortionCoeffs_f8.f1[4]),
                                              static_cast<float4>(distortionCoeffs_f8.f1[5]),
                                              static_cast<float4>(distortionCoeffs_f8.f1[6]),
                                              static_cast<float4>(distortionCoeffs_f8.f1[7])};
```